### ISSUE TITLE
Is there really big difference with Ubuntu and Windows plotting times?

### ISSUE THREAD

**Issue by jussipvainionpaa** [madMAx43v3r/chia-plotter#696]
So I had two different plotter machines and here are my results:

**Plotter #1**
Ryzen 9 3900X
128GB RAM 3200MHz
Samsung 970 Evo Plus 1TB
OS: Ubuntu Mate 20.04
_Plotting time using ramdisk: 27 minutes_

**Plotter #2**
Ryzen 9 3900X (same model but different chip)
128GB RAM 3200MHz (exact same sticks as in Plotter #1)
x4 Samsung 980 PRO 500GB in RAID0 (Striped volume)
OS: Windows 10
_Plotting time using ramdisk: 33 minutes_

The 970 Evo was also the boot drive in Plotter #1 and Plotter #2 had SATA SSD as a boot drive.
I did do a CrystalMark test on the 980 PROs to see that the striped volume was actually working and I did get 12000 MB/s read and 10000 MB/s write speeds.

I don't understand why the Plotter #2 setup is slower when it has otherwise same hardware besides OS.
Or does the boot drive speed have any effect on plotting speed? Should I boot from M.2 on Windows as well?
I'm going to install Ubuntu Mate on another SATA SSD today to try if it makes a big difference.

There's also the fact that I have two different Ryzen 3900X's which I'm using so it could be just the chips performing differently but so much? They're both running without any overclocking.

**Comment by mehditlili** [comment]
This is a question for chiaforum.com

**Comment by ataa** [comment]
I am using ubuntu 21.04 minimal and I get 10% better performance than 20.04 server on the same hardware. My boot driver is USB 2.0 flash drive, I tried booting from ssd but the results were the same as USB 2.0! So switched back to USB, now I have one more sata port to attach extra drive.

**Comment by jussipvainionpaa** [comment]
Wow, moving from Windows 10 to Ubuntu Mate 20.04 dropped plotting time to 26 minutes.
I did change filesystem of the raid 0 array to XFS from NTFS also.

26 minutes is still not what I was expecting when moving from a single 970 EVO PLUS 1 TB to four 980 PRO 500GB drives...
Well at least a single plot is now only taking 0.1 TBW from my drives so I can make 3000 plots before the warranty is up.

I'm thinking if this is a cpu bottleneck I'm facing right now. I'm getting a 5950X to replace my 3900X in the next few weeks so fingers crossed I'll be getting below 20 minute plots soon.

**Comment by FlexiMiners** [comment]
> 
> 
> Wow, moving from Windows 10 to Ubuntu Mate 20.04 dropped plotting time to 26 minutes.
> I did change filesystem of the raid 0 array to XFS from NTFS also.
> 
> 26 minutes is still not what I was expecting when moving from a single 970 EVO PLUS 1 TB to four 980 PRO 500GB drives...
> Well at least a single plot is now only taking 0.1 TBW from my drives so I can make 3000 plots before the warranty is up.
> 
> I'm thinking if this is a cpu bottleneck I'm facing right now. I'm getting a 5950X to replace my 3900X in the next few weeks so fingers crossed I'll be getting below 20 minute plots soon.

I just hit 14.3 min per plot, but there seems to be a thread scalability limit somewhere in the testing I have done, with 2x 28C CPUs and PCIeGen4 Optane SSDs as well as Large RAMdisk

**Comment by digitalscream** [comment]
Yes, it makes a big difference, even with the default kernel options and setup.

There are a _lot_ more optimisations you can make with Linux that just aren't available on Windows, too. Some examples...

- Formatting RAID tmp drives with XFS, configured with proper sunit and swidth for the array
- Disabling kernel mitigations for Spectre/Meltdown/etc
- Forcing the CPU governor to "performance" so it's always boosting

**Comment by jussipvainionpaa** [comment]
> Yes, it makes a big difference, even with the default kernel options and setup.
> 
> There are a _lot_ more optimisations you can make with Linux that just aren't available on Windows, too. Some examples...
> 
>     * Formatting RAID tmp drives with XFS, configured with proper sunit and swidth for the array
> 
>     * Disabling kernel mitigations for Spectre/Meltdown/etc
> 
>     * Forcing the CPU governor to "performance" so it's always boosting

Oh nice. Thanks for the optimization tips, I'll look into them if I can squeeze out even more minutes.
